app/src/test/kotlin/compiler/e2e/TypeCheckingErrorsTest.kt

```diff
@@ -1,5 +1,927 @@
 package compiler.e2e
 
+import compiler.common.diagnostics.Diagnostic
+import compiler.e2e.E2eAsserter.assertErrorOfType
+import compiler.e2e.E2eAsserter.assertProgramCorrect
+import org.junit.Ignore
+import org.junit.Test
+
 class TypeCheckingErrorsTest {
-    // TODO: implement tests for type checking errors
+    private fun assertInvalidTypeError(program: String) {
+        assertErrorOfType(program, Diagnostic.InvalidType::class)
+    }
+
+    private fun assertConditionalMismatchError(program: String) {
+        assertErrorOfType(program, Diagnostic.ConditionalTypesMismatch::class)
+    }
+
+    private fun assertUninitializedGlobalVariableError(program: String) {
+        assertErrorOfType(program, Diagnostic.UninitializedGlobalVariable::class)
+    }
+
+    private fun assertConstantWithoutValueError(program: String) {
+        assertErrorOfType(program, Diagnostic.ConstantWithoutValue::class)
+    }
+
+    private fun assertNonConstantExpressionError(program: String) {
+        assertErrorOfType(program, Diagnostic.NonConstantExpression::class)
+    }
+
+    @Ignore
+    @Test
+    fun `test define variable with unit type`() {
+        assertInvalidTypeError("czynność test() { zm a: Nic; }")
+        assertInvalidTypeError("czynność test() { wart a: Nic; }")
+        assertInvalidTypeError("czynność test() { stała a: Nic; }")
```

Comment:
Sorry I didn't write it before, but I think variables and properties with type `Nic` are legal after all: they have only one possible value `nic`. Could we remove this test, as well as the next one?